File: src/test/java/bad/robot/radiate/ui/FrameRateTest.java
package bad.robot.radiate.ui;

import org.junit.Test;

import static bad.robot.radiate.ui.FrameRate.framesPerSecond;
import static com.googlecode.totallylazy.matchers.Matchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class FrameRateTest {

    @Test
    public void framesPerSecondAsMillisecondFrequency() {
        assertThat(framesPerSecond(24).asFrequencyInMillis(), is(41));
        assertThat(framesPerSecond(25).asFrequencyInMillis(), is(40));
        assertThat(framesPerSecond(30).asFrequencyInMillis(), is(33));
    }

    @Test (expected = IllegalArgumentException.class)
    public void tooHigh() {
        framesPerSecond(61);
    }

    @Test (expected = IllegalArgumentException.class)
    public void tooLow() {
        framesPerSecond(0);
    }


}

File: src/test/scala/bad/robot/radiate/ui/FrameRateTest.scala
package bad.robot.radiate.ui

import bad.robot.radiate.ui.FrameRate._
import org.specs2.mutable.Specification

class FrameRateTest extends Specification {

  "Frames per second as millisecond frequency" >> {
    framesPerSecond(24).asFrequencyInMillis must_== 41
    framesPerSecond(25).asFrequencyInMillis must_== 40
    framesPerSecond(30).asFrequencyInMillis must_== 33
  }

  "too high" >> {
    framesPerSecond(61) must throwA[IllegalArgumentException]
  }

  "too low" >> {
    framesPerSecond(0) must throwA[IllegalArgumentException]
  }

}
